rsa/rsa_factory.go

```diff
@@ -12,60 +12,49 @@
 package rsa
 
 import (
-	"bytes"
 	"crypto/rand"
 	"crypto/rsa"
 	"crypto/sha256"
 	"crypto/x509"
-	"encoding/asn1"
-	"encoding/pem"
-	"fmt"
+	"encoding/base64"
 )
 
 // GenerateRSAKeyPair generates a RSA key pair for the provided bit size.
-// Recommended bit size is 4096
-func GenerateRSAKeyPair(bitSize int) (*pem.Block, *pem.Block, error) {
+func GenerateRSAKeyPair() (*rsa.PrivateKey, error) {
 	reader := rand.Reader
+	return rsa.GenerateKey(reader, 4096)
+}
```

Comment:
can we move the size to the function parameters, actually I was planning to use 2048 bits, has that being deprecated already?

Reply:
The point of hardcoding it is to keep it the same across all plugins